test/plugin_functional/test_suites/core_plugins/rendering.ts

```diff
@@ -120,8 +121,82 @@ export default function({ getService, getPageObjects }: PluginFunctionalProvider
 
       await loaded();
 
-      expect(await testSubjects.exists('coreLegacyCompatH1')).to.be(true);
-      expect(await testSubjects.exists('renderingHeader')).to.be(false);
+      expect(await exists('coreLegacyCompatH1')).to.be(true);
+      expect(await exists('renderingHeader')).to.be(false);
+    });
+
+    it('navigates between standard application and one with custom appRoute', async () => {
+      await navigateTo('/');
+
+      const loaded = await init();
+
+      await loaded();
+
+      async function navigateToApp(title: string) {
+        await appsMenu.clickLink(title);
+        return browser.execute(() => {
+          if (!('__RENDERING_SESSION__' in window)) {
+            window.__RENDERING_SESSION__ = [];
+          }
+
+          window.__RENDERING_SESSION__.push(window.location.pathname);
+        });
+      }
```

Comment:
Duplicated code. it can be extracted outside of the test block.

Also NIT: I think the `__RENDERING_SESSION__ ` on browser side could be avoided by keeping a list on the server side and having the `browser.execute` return the pathname that could then be added to the server-side list. Would also avoid the `const session = await browser.execute` block as the list would be server-side.

Reply:
Good catch on the duplicated code, I'll fix that. Lemme see what I can come up with regarding the paths.

Reply:
Ah, now I remember why I did this. I wanted to keep a variable in the page as proof that the page was not reloaded. I'll add some comments to make that clear.